pkg/clustermanager/cluster_manager.go

```diff
@@ -56,10 +56,14 @@ func GetRegisteredClusters(c client.Client) ([]types2.Cluster, error) {
 		return nil, errors.Wrapf(err, "failed to get clusterSecret secrets")
 	}
 	for _, clusterSecret := range secrets.Items {
+		endpoint := string(clusterSecret.Data["endpoint"])
+		if endp, ok := clusterSecret.GetLabels()[v1alpha1.LabelKeyClusterEndpointType]; ok {
+			endpoint = endp
```

Comment:
Why do we have two sources for `endpoint`?

Reply:
ClusterGateway undergoes a version update which means we need to support backward-compatibility.